cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamRecordReader.java

```diff
@@ -49,10 +58,14 @@
    *
    * @param decoder The decoder to use for decoding stream events.
    */
-  StreamRecordReader(StreamEventDecoder<K, V> decoder) {
+  StreamRecordReader(StreamEventDecoder<K, V> decoder, AuthorizationEnforcer authorizationEnforcer,
+                     AuthenticationContext authenticationContext, StreamId streamId) {
     this.decoder = decoder;
     this.events = Lists.newArrayListWithCapacity(1);
     this.currentEntry = new StreamEventDecoder.DecodeResult<>();
+    this.authorizationEnforcer = authorizationEnforcer;
+    this.principal = authenticationContext.getPrincipal();
```

Comment:
just checking: the record reader is created for every MR job, so its ok to cache the principal? Maybe add a note again